### PR TITLE
Stork Quickstart

### PR DESCRIPTION
**Pull request by cescoffier**
Related to https://github.com/quarkusio/quarkus/pull/22592

**Checklist**:

Your pull request:

- [X] targets the `development` branch
- [X] uses the `999-SNAPSHOT` version of Quarkus
- [X] has tests (`mvn clean test`)
- [X] works in native (`mvn clean package -Pnative`)
- [X] has native tests (`mvn clean verify -Pnative`)
- [X] makes sure the associated guide must not be updated
- [X] links the guide update pull request (if needed)
- [X] updates or creates the `README.md` file (with build and run instructions)
- [X] for new quickstart, is located in the directory _component-quickstart_
- [X] for new quickstart, is added to the root `pom.xml` and `README.md`

### CODE REVIEW

File: stork-quickstart/pom.xml [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+      <artifactId>quarkus-rest-client-reactive</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-smallrye-stork</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>io.smallrye.stork</groupId>
+      <artifactId>smallrye-stork-service-discovery-consul</artifactId>
+      <version>1.0.0.Beta1</version>
```
Comment (gsmet) [inline stork-quickstart/pom.xml:40]: Again, maybe let's add these versions to the BOM?

File: stork-quickstart/pom.xml [hunk 2]
```diff
@@ -0,0 +1,140 @@
+<?xml version="1.0"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
```
Comment (gsmet) [inline stork-quickstart/pom.xml:2]: Could you reformat with 4 spaces? That's what we're trying to do with your POMs.

File: rest-client-reactive-quickstart/src/main/resources/application.properties [hunk 3]
```diff
@@ -1,2 +1,4 @@
 quarkus.tls.trust-all=true
 quarkus.rest-client.extensions-api.url=https://stage.code.quarkus.io/api
+quarkus.rest-client.extensions-api.shared=true
+quarkus.rest-client.extensions-api.name=fojooh
```
Comment (gsmet) [inline rest-client-reactive-quickstart/src/main/resources/application.properties:4]: Is this normal?

**Review by gsmet — COMMENTED** [review]
Added a few comments and questions.

File: rest-client-reactive-quickstart/src/main/resources/application.properties (hunk 3 above)
Comment (michalszynkiewicz) [reply rest-client-reactive-quickstart/src/main/resources/application.properties:4]: sharing is an advanced feature IMO and I don't think this should be added to the quickstart

**Comment by michalszynkiewicz** [comment]
I think it would be good to add a command to start Consul somewhere. Maybe some script or docker-compose

File: stork-quickstart/pom.xml (hunk 1 above)
Comment (cescoffier) [reply stork-quickstart/pom.xml:40]: Waiting for Stork 1.0 to do so. It should not be long...

File: rest-client-reactive-quickstart/src/main/resources/application.properties (hunk 3 above)
Comment (cescoffier) [reply rest-client-reactive-quickstart/src/main/resources/application.properties:4]: Left-overs from another branch.

**Comment by cescoffier** [comment]
All done (reformated the pom and remove the rest client shared attribute).

**Comment by gsmet** [comment]
@michalszynkiewicz I let you do a final check and merge it if all is good.

**Pull request merged**